### ISSUE TITLE
Remotecommand doesn't work

### ISSUE THREAD

**Issue by pavel-hushcha** [warpdotdev/Warp#1761]
### Local Shell and Version

zsh 5.8.1

### Remote Shell and Version

bash 5.0.17(1)-release

### Operating System and Version

MacOs 12.5.1

### Warp Version

v0.2022.08.23.08.06.stable_01

### I have tried commenting out my system & user rcfiles and warp still will not bootstrap

- [X] Yes

### Screenshots

![image](https://user-images.githubusercontent.com/73911482/186829179-0ca24f5b-9d5c-47ae-b893-100ebc679a35.png)
![image](https://user-images.githubusercontent.com/73911482/186829290-01041272-123e-4b2b-b951-d36bd2a6381d.png)

### Include shell xtrace output

![image](https://user-images.githubusercontent.com/73911482/186829445-ebf84452-22e5-4cfa-968e-202825e7dad5.png)

**Comment by elviskahoro** [comment]
@pavel-hushcha will reach out to you when we have updates on this issue.

**Comment by alessandro-newzoo** [comment]
@elviskahoro now that a year and a half has passed, has there been any update on this please?
[This update from March 2022](https://github.com/warpdotdev/Warp/issues/578#issuecomment-1078458506) was supposed to introduce support for `RemoteCommand`:
> SSH now supports LocalCommand / RemoteCommand

but as of today it still won't work:

```
Host test
    HostName <redacted_ip>
    User master_jryhrxuqqz
    RemoteCommand cd /home/master/applications/live/public_html
```

<img width="1023" alt="image" src="https://github.com/warpdotdev/Warp/assets/47320294/b407301b-99b6-4066-9238-1eb28d610e93">

**Comment by ElijahLynn** [comment]
duplicate: https://github.com/warpdotdev/Warp/issues/4466

**Comment by ElijahLynn** [comment]
I suspect the way the ssh wrapper works is to use it's own RemoteCommand. There would have to be a way to either run two RemoteCommand's or merge the two somehow.

**Comment by ElijahLynn** [comment]
related discussion: 
* https://github.com/warpdotdev/Warp/discussions/502#discussioncomment-2001766

**Comment by scubbo** [comment]
Confirming I'm also encountering this on `v0.2024.07.09.08.01.stable_00`

**Comment by ddfisher** [comment]
Hi all -- we've been working hard on a new approach to ssh that should fix this bug (and a number of others!). We'd love your feedback!

If you're up for giving it a shot, here's how you can try it out:
* Update to the latest version (`v0.2024.12.03.08.02.stable_04`).
* Enable the new ssh feature:
    * On macOS, run `defaults write dev.warp.Warp-Stable SshTmuxWrapperOverride true`.
    * On Linux, add the `"SshTmuxWrapperOverride": true` key-value pair to `~/.config/warp-terminal/user_preferences.json`.
* Quit and re-open Warp.

**Comment by ElijahLynn** [comment]
Hi, 

Thanks for the update and for working on fixing this. 

I followed the above instructions and am still getting:

```
ssh host1 "whoami"
Cannot execute command-line and remote command.
```


v0.2024.12.03.08.02.stable_04
![image](https://github.com/user-attachments/assets/7e183736-8225-45d1-926e-65e902913a90)


```
defaults read dev.warp.Warp-Stable SshTmuxWrapperOverride
true
```

In addition, now I am getting the warpify suggestion on every SSH login, and if I hit escape to bypass, it asks me again the next time. 

![image](https://github.com/user-attachments/assets/eb99c760-3d31-46ff-865f-c68f2d410e02)

**Comment by ddfisher** [comment]
Thanks for giving this a try!  A few things:
- Hmm -- that's strange! I can't reproduce the problem myself unless I set a `RemoteCommand` for the host in my ssh config -- do you have a `RemoteCommand` set for `host1` in `~/.ssh/config`, by any chance?
- Sorry about the repeated prompts: we have a "never ask again for this host" button coming next week.
- We won't install anything on the remote without explicitly asking first! If the remote already has `tmux` installed, then we won't need to install anything.  If it doesn't, then we'll offer to install it for you (which you can decline).

**Comment by ElijahLynn** [comment]
Yup, I have a `RemoteCommand` set in ~/.ssh/config. e.g.

```
host1
    RemoteCommand sudo su - newusername
```

**Comment by ddfisher** [comment]
I think that--in this case--that error is unrelated to Warp, actually. From what I've seen, having a `RemoteCommand` set in ssh config is incompatible with providing a program to execute on the command line.

In the old implementation of ssh support, Warp made changes to how ssh was called that were incompatible with having a `RemoteCommand` set in config, resulting in the `Cannot execute command-line and remote command` error. The updated implementation no longer does this.

In summary:
- `ssh host` (with `RemoteCommand` set): Previously incompatible with Warp. Should be fixed by the new approach to ssh.
- `ssh host command_to_run` (with `RemoteCommand` set): Not supported by ssh. Should not work in any terminal.

**Comment by ElijahLynn** [comment]
Thanks for the clarification. And that is interesting that I have been using #1 fine for a while now:
> ssh host (with RemoteCommand set): Previously incompatible with Warp. Should be fixed by the new approach to ssh.
